app/screens/home/channel_list/categories_list/categories/categories.tsx

```diff
@@ -83,24 +96,43 @@ const Categories = ({categories, currentTeamId, onlyUnreads, unreadsOnTop}: Prop
         return orderedCategories;
     }, [categories, onlyUnreads, unreadsOnTop]);
 
+    const categoriesAfterLoading = useRef(categoriesToShow);
+
+    useDidUpdate(() => {
+        const t = setTimeout(() => {
+            categoriesAfterLoading.current = categoriesToShow;
+            setLoading(false);
+        }, 350);
+        return () => clearTimeout(t);
+    }, [categoriesToShow]);
+
     if (!categories.length) {
         return <LoadCategoriesError/>;
     }
 
     return (
-        <FlatList
-            data={categoriesToShow}
-            ref={listRef}
-            renderItem={renderCategory}
-            style={styles.mainList}
-            showsHorizontalScrollIndicator={false}
-            showsVerticalScrollIndicator={false}
-            keyExtractor={extractKey}
-            initialNumToRender={categoriesToShow.length}
-
-            // @ts-expect-error strictMode not included in the types
-            strictMode={true}
-        />
+        <>
+            {!isLoading && (
+                <FlatList
+                    data={categoriesAfterLoading.current}
+                    ref={listRef}
+                    renderItem={renderCategory}
+                    style={styles.mainList}
+                    showsHorizontalScrollIndicator={false}
+                    showsVerticalScrollIndicator={false}
+                    keyExtractor={extractKey}
+                    initialNumToRender={categoriesToShow.length}
+
+                    // @ts-expect-error strictMode not included in the types
+                    strictMode={true}
+                />
+            )}
+            {isLoading && (
+                <View style={styles.loadingView}>
+                    <Loading style={styles.loading}/>
+                </View>
```

Comment:
Added loading for better perceived performance.

Reply:
I would probably move this entire thing one level up, to avoid the queries from executing

Reply:
I differ here. We want to start calculating the categories while the team switch happens, so we have all the categories ready when we start to render. If not, we will have to do on the "same render" the category calculation and the actual rendering.